### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by jon-sully**
## Pull Request

**Summary:**
Still a bit WIP but wanted to pretty significantly revamp the main repo README with the V2 changes

There's still a lot more content to work on, but I at least wanted to get this up!

**Related Issue:**
n/a

**Description:**
V2 changed a lot of stuff! Felt like the whole README needed a big look and revamp ❤️ 

**Testing:**
n/a

**Screenshots (if applicable):**
Recommend reading from scratch instead of reading via diffs: https://github.com/jon-sully/noticed/tree/update-readme

### CODE REVIEW

**Comment by excid3** [comment]
Looking good so far!

File: README.md [hunk 1]
```diff
@@ -38,13 +44,13 @@ Bulk delivery methods (one notification for all recipients):
...
 ## 🚀 Installation
-Run the following command to add Noticed to your Gemfile:
+Run the following command to add Noticed to your Gemfile
 
 ```ruby
 bundle add "noticed"
 ```
 
-Add the migrations:
+Add the migraitons
```
Comment (bvogel) [inline README.md:53]:
```suggestion
Add the migrations
```

Comment (jon-sully) [reply README.md:53]: D'oh! Thought I caught that one. Thanks for the suggestion! 👍

**Comment by jon-sully** [comment]
One thing I'll ask for clarity on, @excid3. I don't recall if it was required or optional, but we used explicit `params` in my app's V1 Notifications. In V2 explicit params are totally optional and declared instead under the macro `required_params` (or singular `required_param`) instead of just `params`, right?

```ruby
class NewCommentNotifier < Noticed::Event
  deliver_by :email do |config|
    config.mailer = "CommentMailer"
		config.if = ->(recipient) { !!recipient.preferences[:email] }
  end

  required_params :record, :foo
end
```

**Comment by jon-sully** [comment]
Planning on wrapping this up today, btw!

**Comment by excid3** [comment]
Sounds good. 👍

`params` from v1 was just renamed to `required_params` for clarity. They both trigger validation to check and ensure the param name was in the `params` hash.

**Comment by jon-sully** [comment]
I think I'm about done here! Massive changes to the README, a few tweaks to the actual library code as prompted by the README.. overall a good turnaround!

**Comment by excid3** [comment]
I'll give it a review after the kiddo goes to bed. 👍

File: README.md [hunk 2]
```diff
@@ -468,3 +510,11 @@ DATABASE_URL=postgres://127.0.0.1/noticed_test rails test
...
 ## 📝 License
 The gem is available as open source under the terms of the [MIT License](https://opensource.org/licenses/MIT).
+
+
+
+
+
+
+
+Search the whole doc for `notification`
```
Comment (bvogel) [inline README.md:518]: I think this is a WIP leftover?

Comment (jon-sully) [reply README.md:518]: D'oh! Yes it is. Thanks!

**Pull request merged**